Makefile.docker

```diff
@@ -0,0 +1,95 @@
+REPO := logdna-agent-v2
+
+DOCKER := docker
+DOCKER_PRIVATE_IMAGE := us.gcr.io/logdna-k8s/logdna-agent-v2
+DOCKER_PUBLIC_IMAGE := docker.io/logdna/logdna-agent
+DOCKER_IBM_IMAGE := icr.io/ext/logdna-agent
+
+# The image repo and tag can be modified e.g.
+# `make -f Makefile.docker build IMAGE=docker.io/rust:1.42.0`
+IMAGE_REPO ?= docker.io/rust
+IMAGE_TAG ?= 1.42
+IMAGE ?= $(IMAGE_REPO):$(IMAGE_TAG)
+IMAGE := $(IMAGE)
+
+VCS_REF := $(shell git rev-parse --short HEAD)
+VCS_URL := https://github.com/logdna/logdna-agent-v2
+BUILD_TIMESTAMP := $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+BUILD_VERSION := $(shell sed -nE "s/^version = \"(.+)\"\$$/\1/p" bin/Cargo.toml)
+BUILD_TAG := $(VCS_REF)
```

Comment:
I opted to use only the VCS_REF as the build tag instead of the ref+timestamp.

There's edge cases with a timestamp when the Makefile is called on multiple times where each call might get a different timestamp. The image should also have a timestamp metadata baked into it if we need it.